internal/config/profile.go

```diff
@@ -226,20 +229,54 @@ func (p *profile) SetOrgID(v string) {
 }
 
 // GetConfigDescription returns a map describing the configuration
-func GetConfigDescription(name string) map[string]string {
-	redactedValue := "redacted"
+func GetConfigDescription() map[string]string { return p.GetConfigDescription() }
+func (p *profile) GetConfigDescription() map[string]string {
+	settings := viper.GetStringMapString(p.Name())
+	newSettings := make(map[string]string)
+
+	for k, v := range settings {
+		if k == privateAPIKey || k == publicAPIKey {
+			newSettings[k] = "redacted"
+		} else {
+			newSettings[k] = v
+		}
+	}
+
+	return newSettings
+}
 
-	m := viper.GetStringMapString(name)
+// Delete deletes an existing configuration
+func Delete() error { return p.Delete() }
+func (p *profile) Delete() error {
+	// Configuration needs to be deleted from toml, as viper doesn't support this yet.
+	// FIXME :: change when https://github.com/spf13/viper/pull/519 is merged.
+	configurationAfterDelete := viper.AllSettings()
 
-	if _, ok := m[privateAPIKey]; ok {
-		m[privateAPIKey] = redactedValue
+	t, err := toml.TreeFromMap(configurationAfterDelete)
+	if err != nil {
+		return err
 	}
 
-	if _, ok := m[publicAPIKey]; ok {
-		m[publicAPIKey] = redactedValue
+	// Delete from the toml manually
+	err = t.Delete(p.Name())
+	if err != nil {
+		return err
+	}
+
+	s := t.String()
+
+	flags := os.O_CREATE | os.O_TRUNC | os.O_WRONLY
+	f, err := p.fs.OpenFile(fmt.Sprintf("%s/%s.toml", p.configDir, ToolName), flags, 0600)
+	if err != nil {
+		return err
+	}
+
+	if _, err := f.WriteString(s); err != nil {
+		return err
 	}
 
-	return m
+	// Force reload, so that viper has the new configuration
+	return Load()
```

Comment:
[q] do we need this? if yes you may want to use `p.Load()`

Reply:
It'd be more correct if we were to do anything after deleting a profile, otherwise the old configuration would still be cached. I guess that's a judgement call though -- would be quicker without it and probably won't happen
+1 on p.load instead

Reply:
up to you here, I don't see many people deleting their profiles so speed is not my concern for this one

Reply:
just add to the comment of the method that the profile will be reloaded as we don't do this for other methods